source/crud/tests/v2/find-allowdiskuse.yml

```diff
@@ -0,0 +1,50 @@
+runOn:
+  - { minServerVersion: "4.4.0" }
+
+collection_name: &collection_name 'test_find_allowdiskuse'
+
+tests:
+  -
+    description: "Find does not send allowDiskuse when unspecified"
+    operations:
+      -
+        object: collection
+        name: find
+        arguments:
+          filter: { }
+    expectations:
+      -
+        command_started_event:
+          command:
+            find: *collection_name
+            allowDiskUse:
```

Comment:
The intended meaning of this line is that `allowDiskUse` is NOT present in the command.

Reply:
Sorry to tack on work, but I believe this is the first time we're asserting the absence of a field in a command body within the CRUD spec test suite. It may require some additional clarification in the CRUD test README.

The CRUD spec README currently only refers to the Command Monitoring spec when we talk about the structure of the `command_started_event` field. CRUD's section for [Evaluating Matches](https://github.com/mongodb/specifications/tree/master/source/crud/tests#evaluating-matches) doesn't discuss any special handling for `null` fields in a `result` or `command_started_event` field within the test. As-is, a CRUD test runner would likely assert that the command contains an `allowDiskUse` field with a value of `null`; rather than the desired outcome of asserting that the command document does not contain an `allowDiskUse` field.

I think this warrants adding a section to the CRUD test README that instructs test runners to use matching logic from the Transaction spec test runner for _events_ ([Null Values](https://github.com/mongodb/specifications/blob/master/source/transactions/tests/README.rst#null-values) in particular). For prior art, you might just copy the section from the [Transactions Convenient API](https://github.com/mongodb/specifications/blob/master/source/transactions-convenient-api/tests/README.rst#command-started-events)) test runner.

----

Alternatively, we could remove this test case and punt on this until after [Unified test format](https://jira.mongodb.org/browse/DRIVERS-709) is implemented, since that should address the inconsistent handling of `command_started_event` assertions throughout various specs.

Reply:
Since we've used `{ field : null }` before (e.g. in the transactions tests) to mean that "field" is not present, let's use that for now.

When we do the unified test format that effort should standardize how a JSON test specifies that a field is NOT present.

I think we should have some meta notation like:

    { field : { $exists : false } }
    { field : { $exists : true } }

but that will be decided as part of the Unified Test Format work.

Reply:
I'm fine with keeping `null` here, but I'd still request we add clarification to the test README so this is clearly communicated to implementing drivers. See my "I think this warrants adding a section..." paragraph in my last comment. There should be some existing text you can copy in.

Reply:
There is: https://github.com/mongodb/specifications/blob/master/source/transactions/tests/README.rst#null-values

Reply:
I've added some text about this to `tests/README.RST`

Reply:
From my perspective, this thread can now be resolved.